storybookjs/storybook
# fix: align behavior of name and storyName properties

Closes #22540



## What I did


Update `name` and `storyName` to have consistent behavior with each other and across the sidebar and docs. Both properties will display the same results when assigned to during or after the instantiation of an exported variable (story).

I didn't add any more documentation on this as it is not encouraged to use `storyName` in a CSF3 object, and the user is already warned about this when used directly in a CSF3 object. Other changes did not require updated documentation either.



## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Update Button.stories.ts to include:
```
export const PrimaryNoReset: Story = {
  args: {
    primary: true,
    label: 'Button',
  },
  name: 'Primary',
};

export const PrimaryReset: Story = {
  args: {
    primary: true,
    label: 'Button',
  },
};
PrimaryReset.name = 'Primary (reset)';

export const SecondaryNoReset: Story = {
  args: {
    label: 'Button',
  },
  storyName: 'Secondary',
};

export const SecondaryReset: Story = {
    args: {
      label: 'Button',
    },
};
SecondaryReset.storyName = 'Secondary (reset)';
```
3. Open Storybook in your browser
4. Check name of stories in sidebar and docs

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
